Sources/SkipBluetooth/CBCentralManager.swift

```diff
@@ -0,0 +1,269 @@
+    private let bondingReceiver: BondCallback
+
+    // TODO: Allow multiple connections at a time
+    private var pendingGatts: BluetoothGatt?
+
+    private var scanner: BluetoothLeScanner? {
+        adapter?.getBluetoothLeScanner()
+    }
+
+    public var delegate: (any CBCentralManagerDelegate)? {
+        get {
+            gattDelegate.centralManagerDelegate
+        } set {
+            scanDelegate.delegate = newValue
+            gattDelegate.centralManagerDelegate = newValue
+        }
+    }
+
+    public var isScanning: Bool { adapter?.isDiscovering() ?? false }
+
+    public override init() {
+        super.init()
+        stateChangedHandler = {
+            delegate?.centralManagerDidUpdateState(self)
+        }
+        bondingReceiver = BondCallback { device in
+            tryConnect(to: device)
+        }
+
+        let filter = IntentFilter(BluetoothDevice.ACTION_BOND_STATE_CHANGED)
+        let context = ProcessInfo.processInfo.androidContext
+        context.registerReceiver(bondingReceiver, filter)
+    }
+
+    open func scanForPeripherals(withServices serviceUUIDs: [CBUUID]?, options: [String : Any]? = nil) {
+        guard hasPermission(android.Manifest.permission.BLUETOOTH_SCAN) else {
+            logger.error("CBCentralManager.scanForPeripherals: Missing BLUETOOTH_SCAN permission.")
+            return
+        }
+
+        let settingsBuilder = ScanSettings.Builder()
+            .setScanMode(ScanSettings.SCAN_MODE_BALANCED)
+            .setCallbackType(ScanSettings.CALLBACK_TYPE_FIRST_MATCH)
+        let filterBuilder = ScanFilter.Builder()
+
+        if let serviceUUIDs = serviceUUIDs {
+            for uuid in serviceUUIDs {
+                filterBuilder.setServiceUuid(ParcelUuid(uuid.kotlin()))
+            }
+        }
+
+        if let isDuplicate = options?[CBCentralManagerScanOptionAllowDuplicatesKey] as? Bool {
+            settingsBuilder.setCallbackType(
+                isDuplicate ? ScanSettings.CALLBACK_TYPE_ALL_MATCHES : ScanSettings.CALLBACK_TYPE_FIRST_MATCH
+            )
+        }
+
+        // SKIP NOWARN
+        if let uuids = options?[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] as? [CBUUID] {
+            for uuid in uuids {
+                filterBuilder.setServiceSolicitationUuid(ParcelUuid(uuid.kotlin()))
+            }
+        }
+
+        let settings = settingsBuilder.build()
+        let scanFilters = listOf(filterBuilder.build())
+
+        scanner?.startScan(scanFilters, settings, scanDelegate)
+        logger.info("CBCentralManager.scanForPeripherals: Starting Scan")
+    }
+
+    public func stopScan() {
+        guard hasPermission(android.Manifest.permission.BLUETOOTH_SCAN) else {
+            logger.error("CBCentralManager.scanForPeripherals: Missing BLUETOOTH_SCAN permission")
+            return
+        }
+
+        logger.info("CentralManager.stopScan: Stopping Scan")
+        scanner?.stopScan(scanDelegate)
+    }
+
+#if !SKIP
```

Comment:
You have a number of #if !SKIP blocks, but the whole thing is surrounded by #if SKIP, so they aren't functional. This pattern is repeated in other files. Is this your way of excluding things that aren't implemented? If so, why not use @available(*, unavailable) for these? Do the signatures use types that aren't implemented or similar?

Sometimes in those cases we change the unimplemented types to Any (i.e. func f(x: Any /* OriginalType */) and mark @available(*, unavailable) to hopefully give a better error message on attempted use. But fine either way - just checking that I understand.

Reply:
These are remnants from my initial implementation. I've replaced these sites with `@available(*, unavailable)` references instead. Sorry for the confusion here.